### PR TITLE
Filled regions 2016

### PR DESCRIPTION
**Pull request by moethu**
### Purpose

Added Revit FilledRegions and FilledRegionTypes to Dynamo

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [x] Snapshot of UI changes, if any.

### Reviewers

@mjkkirschner 
The FilledRegionTypes depend on the Dropdowns commited earlier with rooms.

### FYIs

@sharadkjaiswal @Randy-Ma

### CODE REVIEW

File: src/Libraries/RevitNodes/Elements/FilledRegion.cs [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+        /// <param name="boundary">Boundary curves</param>
+        /// <param name="regionType">Region Type</param>
+        /// <returns></returns>
+        public static FilledRegion ByCurves(Revit.Elements.Views.View view, IEnumerable<Autodesk.DesignScript.Geometry.Curve> boundary, FilledRegionType regionType)
+        {
+            Autodesk.Revit.DB.FilledRegionType type = regionType.InternalRevitElement;
+
+            CurveLoop loop = new CurveLoop();
+            foreach (Autodesk.DesignScript.Geometry.Curve curve in boundary)
+                loop.Append(curve.ToRevitType());
```
Comment (mjkkirschner) [inline src/Libraries/RevitNodes/Elements/FilledRegion.cs]: do you need to pass true here to convert with host units? I cannot remember the default?

Comment (mjkkirschner) [inline src/Libraries/RevitNodes/Elements/FilledRegion.cs]: also, should we do some checking to see if this forms a loop?

File: src/Libraries/RevitNodes/Elements/FilledRegionType.cs [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+using System.Collections.Generic;
+
+namespace Revit.Elements
+{
+    /// <summary>
+    /// Revit filled Region Type
+    /// </summary>
+    [DynamoServices.RegisterForTrace]
+    public class FilledRegionType : Element
+    {
```
Comment (mjkkirschner) [inline src/Libraries/RevitNodes/Elements/FilledRegionType.cs]: no need for any public properties on this type?

File: test/Libraries/RevitNodesTests/Elements/FilledRegionTests.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            FilledRegionType type = FilledRegionType.FromExisting((Autodesk.Revit.DB.FilledRegionType)collector.FirstOrDefault(), true);
+
+            Assert.NotNull(type);
+
+            Polygon polygon = Polygon.RegularPolygon(Circle.ByCenterPointRadius(Point.ByCoordinates(0, 0, 0), 10), 4);
+            Assert.NotNull(polygon);
+
+            var reg = FilledRegion.ByCurves(Revit.Application.Document.Current.ActiveView, polygon.Curves().ToList(), type);
+            Assert.NotNull(reg);
+
```
Comment (mjkkirschner) [inline test/Libraries/RevitNodesTests/Elements/FilledRegionTests.cs]: what happens when you use a circle?

**Comment by mjkkirschner** [comment]
@moethu some questions.

File: src/Libraries/RevitNodes/Elements/FilledRegion.cs (hunk 1 above)
Comment (moethu) [inline src/Libraries/RevitNodes/Elements/FilledRegion.cs]: default is true for conversion. good point about the open/closed loop. added a check.

File: src/Libraries/RevitNodes/Elements/FilledRegionType.cs (hunk 2 above)
Comment (moethu) [inline src/Libraries/RevitNodes/Elements/FilledRegionType.cs]: exposed Name, color and Pattern

File: test/Libraries/RevitNodesTests/Elements/FilledRegionTests.cs (hunk 3 above)
Comment (moethu) [inline test/Libraries/RevitNodesTests/Elements/FilledRegionTests.cs]: should be fine, since a circle is defined by two arcs --> closed curve loop.

**Comment by moethu** [comment]
@mjkkirschner changes applied

File: test/Libraries/RevitNodesTests/Elements/FilledRegionTests.cs (hunk 3 above)
Comment (mjkkirschner) [inline test/Libraries/RevitNodesTests/Elements/FilledRegionTests.cs]: can you add a test for creating a region from a circle?

**Comment by mjkkirschner** [comment]
just marking the conflicts here.

**Comment by moethu** [comment]
resolved

**Pull request merged**